src/DonationForms/Rules/CurrencyRule.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Give\DonationForms\Rules;
+
+use Closure;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidationRule;
+
+/**
+ * Custom currency validation rule that uses GiveWP's currency list and filter.
+ * Replaces the generic stellar validation library currency rule to ensure consistency
+ * with GiveWP's configured currencies and filters.
+ * 
+ * @unreleased
+ */
+class CurrencyRule implements ValidationRule
```

Comment:
Could you create a unit test for this rule?  I have a helper for asserting rules you can checkout this one:

https://github.com/impress-org/givewp/blob/c4fb6a0cd5863c13f6ea3fd577a98c363f36c5e6/tests/Unit/DonationForms/Rules/TestHoneyPotRule.php#L15